src/main/java/org/tron/common/runtime/Runtime.java

```diff
@@ -266,10 +424,149 @@ public void go() {
           deposit.commit();
         }
       }
-    } catch (Exception e) {
+    } catch (OutOfResourceException e) {
+      logger.error(e.getMessage());
+      runtimeError = e.getMessage();
+    } catch (TronException e) {
+      spendUsage(false);
       logger.error(e.getMessage());
       runtimeError = e.getMessage();
     }
+    //todo catch over resource exception
+//    catch (Exception e) {
+//      logger.error(e.getMessage());
+//      runtimeError = e.getMessage()
+//  }
+
+  }
+
+  private void spendUsage(boolean spandStorage) {
+    long cpuUsage, storageUsage;
+    storageUsage = 0;
+    long now = System.nanoTime() / 1000;
+    cpuUsage = now - program.getVmStartInUs();
+    if (executerType == ET_NORMAL_TYPE) {
+      /*
+       * trx.getCpuRecipt
+       *
+       * */
+    }
+    ContractCapsule contract = deposit.getContract(result.getContractAddress());
+    ByteString originAddress = contract.getInstance().getOriginAddress();
+    AccountCapsule origin = deposit.getAccount(originAddress.toByteArray());
+
+    byte[] callerAddressBytes = TransactionCapsule.getOwner(trx.getRawData().getContract(0));
+    AccountCapsule caller = deposit.getAccount(callerAddressBytes);
+
+    spendCpuUsage(cpuUsage, origin, caller);
+    if (spandStorage) {
+      spendStorageUsage(storageUsage, origin, caller);
+    }
+  }
+
+  private void spendCpuUsage(long cpuUsage, AccountCapsule origin, AccountCapsule caller) {
+    //TODO get origin cpu
+    long originCpu;
+    originCpu = getCpuByAccount(origin);
+
+    //TODO get caller cpu
+    long callerCpu;
+    callerCpu = getCpuByAccount(caller);
+    //TODO get caller cpulimit（trx）
+    long callerCpuLimit;
+    callerCpuLimit = 0;
+    cpuUsage = getCpuUsageLess(cpuUsage, origin, originCpu);
+    if (cpuUsage <= 0) {
+      return;
+    }
+    cpuUsage = getCpuUsageLess(cpuUsage, caller, callerCpu);
+    if (cpuUsage <= 0) {
+      return;
+    }
+    long overCpu = getCpuByLimit(callerCpuLimit);
+    cpuUsage = getCpuUsageLess(cpuUsage, caller, overCpu);
+    if (cpuUsage <= 0) {
+      return;
+    }
+  }
+
+  private long getCpuUsageLess(long cpuUsage, AccountCapsule origin, long cpu) {
+    if (cpuUsage <= cpu) {
```

Comment:
![Bug](https://sonarsource.github.io/sonar-github/bug.png 'Bug') Bug: Remove this conditional structure or edit its code blocks so that they're not all the same. ([squid:S3923](https://sonarcloud.io/organizations/tron-zhaohong/rules#rule_key=squid%3AS3923))

[See it in SonarCloud](https://sonarcloud.io/project/issues?id=java-tron&pullRequest=1188&issues=AWTgZvGN3lT_iufdZyBE&open=AWTgZvGN3lT_iufdZyBE)

[ISSUE_KEY]: AWTgZvGN3lT_iufdZyBE